### PR TITLE
fix: check if downloads need tar -xvf files

### PR DESCRIPTION
**Pull request by o0OooO**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by NingLu** [comment]
We identified new issues on unchanged lines of code. Navigate to the Amazon CodeGuru Reviewer console to view the recommendations to fix them.

File: aws_extension/mme_utils.py [hunk 1]
```diff
@@ -100,11 +105,24 @@ def upload_model(model_type, model_name, model_s3_pos):
...
-    os.system("df -h")
+    logging.info(tar_name)
+    command = f"file --mime-type -b ./{tar_name}"
+    file_type = subprocess.check_output(command, shell=True).decode('utf-8').strip()
+    logging.info(f"file_type is {file_type}")
+    if file_type == TAR_TYPE_FILE:
+        logging.info("model type is tar")
+        os.system(f"tar xvf {tar_name}")
+        os.system(f"rm {tar_name}")
+        os.system("df -h")
```
Comment (NingLu) [inline aws_extension/mme_utils.py:119]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a shell: Seems safe, but may be changed in the future, consider rewriting without shell https://bandit.readthedocs.io/en/latest/plugins/b605_start_process_with_a_shell.html

File: aws_extension/mme_utils.py [hunk 2]
```diff
@@ -100,11 +105,24 @@
...
     #download from s3
+    logging.info(f'./tools/s5cmd cp {model_s3_pos} ./')
     os.system(f'./tools/s5cmd cp {model_s3_pos} ./')
     tar_name = model_s3_pos.split('/')[-1]
-    os.system(f"tar xvf {tar_name}")
-    os.system(f"rm {tar_name}")
-    os.system("df -h")
+    logging.info(tar_name)
+    command = f"file --mime-type -b ./{tar_name}"
+    file_type = subprocess.check_output(command, shell=True).decode('utf-8').strip()
```
Comment (NingLu) [inline aws_extension/mme_utils.py:113]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

subprocess call with shell=True identified, security issue. https://bandit.readthedocs.io/en/latest/plugins/b602_subprocess_popen_with_shell_equals_true.html

File: aws_extension/mme_utils.py [hunk 3]
```diff
@@ -100,11 +105,24 @@
...
+    if file_type == TAR_TYPE_FILE:
+        logging.info("model type is tar")
+        os.system(f"tar xvf {tar_name}")
+        os.system(f"rm {tar_name}")
+        os.system("df -h")
+    else:
+        os.system(f"rm {tar_name}")
+        logging.info(f"model type is origin file type: {file_type}")
+        prefix_name = model_s3_pos.split('.')[0]
+        os.system(f'./tools/s5cmd cp {prefix_name}* ./models/{model_type}/')
```
Comment (NingLu) [inline aws_extension/mme_utils.py:124]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a shell, possible injection detected, security issue. https://bandit.readthedocs.io/en/latest/plugins/b605_start_process_with_a_shell.html

File: aws_extension/mme_utils.py [hunk 4]
```diff
@@ -100,11 +105,24 @@
...
-    os.system("df -h")
+    logging.info(tar_name)
+    command = f"file --mime-type -b ./{tar_name}"
+    file_type = subprocess.check_output(command, shell=True).decode('utf-8').strip()
+    logging.info(f"file_type is {file_type}")
+    if file_type == TAR_TYPE_FILE:
+        logging.info("model type is tar")
+        os.system(f"tar xvf {tar_name}")
+        os.system(f"rm {tar_name}")
+        os.system("df -h")
```
Comment (NingLu) [inline aws_extension/mme_utils.py:119]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a partial executable path https://bandit.readthedocs.io/en/latest/plugins/b607_start_process_with_partial_path.html

File: aws_extension/mme_utils.py [hunk 5]
```diff
@@ -100,11 +105,24 @@
...
-    os.system(f"rm {tar_name}")
-    os.system("df -h")
+    logging.info(tar_name)
+    command = f"file --mime-type -b ./{tar_name}"
+    file_type = subprocess.check_output(command, shell=True).decode('utf-8').strip()
+    logging.info(f"file_type is {file_type}")
+    if file_type == TAR_TYPE_FILE:
+        logging.info("model type is tar")
+        os.system(f"tar xvf {tar_name}")
+        os.system(f"rm {tar_name}")
```
Comment (NingLu) [inline aws_extension/mme_utils.py:118]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a shell, possible injection detected, security issue. https://bandit.readthedocs.io/en/latest/plugins/b605_start_process_with_a_shell.html

File: aws_extension/mme_utils.py [hunk 6]
```diff
@@ -100,11 +105,24 @@
...
+        os.system("df -h")
+    else:
+        os.system(f"rm {tar_name}")
```
Comment (NingLu) [inline aws_extension/mme_utils.py:121]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a shell, possible injection detected, security issue. https://bandit.readthedocs.io/en/latest/plugins/b605_start_process_with_a_shell.html

File: aws_extension/mme_utils.py [hunk 7]
```diff
@@ -100,11 +105,24 @@
...
-    os.system(f"tar xvf {tar_name}")
-    os.system(f"rm {tar_name}")
-    os.system("df -h")
+    logging.info(tar_name)
+    command = f"file --mime-type -b ./{tar_name}"
+    file_type = subprocess.check_output(command, shell=True).decode('utf-8').strip()
+    logging.info(f"file_type is {file_type}")
+    if file_type == TAR_TYPE_FILE:
+        logging.info("model type is tar")
+        os.system(f"tar xvf {tar_name}")
```
Comment (NingLu) [inline aws_extension/mme_utils.py:117]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a shell, possible injection detected, security issue. https://bandit.readthedocs.io/en/latest/plugins/b605_start_process_with_a_shell.html

File: test/api_test/txt2img_api_test/txt2img_local_api_user1.py [hunk 8]
```diff
@@ -8,32 +8,37 @@
...
+
+import json
+from _types import InvocationsRequest
 
+def custom_serializer(obj):
+    if isinstance(obj, InvocationsRequest):
+        return obj.__dict__  # 将对象转换为字典
+    raise TypeError("Object not serializable")
+
+request = InvocationsRequest(task='txt2img', username='test',param_s3='s3://stabledeffusion110-sds3aigcbucket7db76a0b-f8h2ckvcafgm/txt2img/infer_v2/068e4cb2-3cb3-41d6-8d37-1d9424940aa3/api_param.json',models={'space_free_size': 40000000000.0, 'Stable-diffusion': [{'s3': 's3://stabledeffusion110-sds3aigcbucket7db76a0b-f8h2ckvcafgm/Stable-diffusion/checkpoint/custom/13fb3cd2-3d7c-41dd-8fcb-78cb8a86297e','id': '13fb3cd2-3d7c-41dd-8fcb-78cb8a86297e','model_name': 'abyssorangemix3AOM3_aom3.safetensors', 'type': 'Stable-diffusion'}]})
```
Comment (alvindaiyan) [inline test/api_test/txt2img_api_test/txt2img_local_api_user1.py:34]: remove s3 info

**Review by alvindaiyan — CHANGES_REQUESTED** [review]

**Comment by alvindaiyan** [comment]
Looks like a group of (safetensor)files shares one checkpoint entry in the ddb table, I'm not sure if this works well and this is NOT how we want save models in the table originally.

File: aws_extension/mme_utils.py [hunk 9]
```diff
@@ -100,11 +105,24 @@ def upload_model(model_type, model_name, model_s3_pos):
...
```
Comment (yike5460) [inline aws_extension/mme_utils.py:117]: It won't work on windows

File: test/api_test/txt2img_api_test/_types.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+# a copy of aws_extensions.models.InvocationsRequest
+@dataclasses.dataclass
+class InvocationsRequest:
+    task: str
+    username: Optional[str]
+    param_s3: str
+    # checkpoint_info:Optional[dict]
+    models: Optional[dict]
+    # txt2img_payload: Optional[StableDiffusionTxt2ImgProcessingAPI]
```
Comment (yike5460) [inline test/api_test/txt2img_api_test/_types.py:13]: Remove unused code

File: test/api_test/txt2img_api_test/txt2img_local_api_user1.py [hunk 11]
```diff
@@ -8,32 +8,37 @@
...
...f0f4c5ce85/cheeseDaddys_41.safetensors', 'AnythingV5Ink_ink.safetensors': 's3://stable-diffusion-aws-extension-aigcbucketa457cb49-xfyck6nj4vlo/Stable-diffusion/checkpoint/custom/1a0227fc-5bb0-436b-aa87-80d487a536b3/AnythingV5Ink_ink.safetensors', 'camelliamix25DV2_v2.safetensors': 's3://stable-diffusion-aws-extension-aigcbucketa457cb49-xfyck6nj4vlo/Stable-diffusion/checkpoint/custom/2f5063ee-e2ac-40be-b48e-8762dfdc25eb/camelliamix25DV2_v2.safetensors', 'sd-v1-5-inpainting.ckpt': 's3://stable-diffusion-aws-extension-aigcbucketa457cb49-xfyck6nj4vlo/Stable-diffusion/checkpoint/custom/822a6754-87e7-495b-b71a-543cf78cefb2/sd-v1-5-inpainting.ckpt', 'yangk-style_2160_lora.safetensors': 's3://stable-diffusion-aws-extension-aigcbucketa457cb49-xfyck6nj4vlo/Stable-diffusion/checkpoint/custom/7a8ad4b0-0159-4c0d-a5b9-a6692f90902a/yangk-style_2160_lora.safetensors'}, 'embeddings': {}, 'Lora': {}, 'hypernetworks': {}, 'ControlNet': {'control_v11p_sd15_canny.pth': 's3://stable-diffusion-aws-extension-aigcbucketa457cb49-xfyck6nj4vlo/ControlNet/checkpoint/custom/a20edd04-535c-4d85-842b-95c3c743d819/control_v11p_sd15_canny.pth'}, 'sagemaker_endpoint': 'infer-endpoint-5d9775d', 'task_type': 'txt2img'}
```
Comment (yike5460) [inline test/api_test/txt2img_api_test/txt2img_local_api_user1.py:20]: Specific bucket name should never exposed

**Review by yike5460 — CHANGES_REQUESTED** [review]

File: aws_extension/mme_utils.py (hunk 9 above)
Comment (o0OooO) [reply aws_extension/mme_utils.py:117]: 这个是在镜像里执行的

**Pull request merged**